### DISCUSSION TITLE
[Android/iOS] Compass heading

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#3308] [Q&A]
Is there a way to calculate compass heading (direction/azimuth) from available AIR classes?
I tried to use `DeviceRotationEvent.yaw`, but it's always set to 0 when app starts so I can't define North direction.

**Comment by ajwfrost** [comment]
Have just looked at the code for this on Android, it should provide the yaw information based on the device rotation information: [TYPE_ROTATION_VECTOR](https://developer.android.com/reference/android/hardware/SensorEvent#sensor.type_rotation_vector:)

Which looks like it should be the actual angle between the phone and magnetic north, although I would assume it's dependent upon the availability of hardware sensors.

I'll ask someone to look at this and see what the device is returning vs what the AS3 API is saying..

**Reply by MalacTheLittle** [reply]
I've tested it on a few different mobile phones, result is the same... magnetic north is unnoted. And Google maps shows heading precise which means hardware sensors are available and works fine.

**Reply by MalacTheLittle** [reply]
Are there any update on this issue?

**Comment by raresn** [comment]
Or you can use this one: https://airnativeextensions.com/extension/com.distriqt.Compass

**Reply by MalacTheLittle** [reply]
I was considering that one, but I think it's strange to use ANE only because of heading when `DeviceRotation` class already reads hardware sensors.

**Comment by MalacTheLittle** [comment]
> I'll ask someone to look at this and see what the device is returning vs what the AS3 API is saying..

@ajwfrost Do you have any news on this issue?
